gpt_mini/utility/data_layer.py

```diff
@@ -0,0 +1,35 @@
+import os
+
+import tensorflow_datasets as tfds
+
+from gpt_mini.utility import logger
+
+log = logger.init("data_layer")
+
+
+class DataLayer:
+    def __init__(self, data_source: str):
+        self._data_source = data_source
+
+    def _get_data(self) -> dict:
+        text_dict = dict()
+        if self._data_source == "local":
+            for split in ["train", "validation", "test"]:
+                local_path = os.path.join(
+                    well_known_paths["DATASETS_DIR"],
+                    f"local-{split}.txt",
+                )
+                try:
+                    with open(local_path, "r") as f:
+                        text_dict[split] = f.read()
+                except:
```

Comment:
Shouldn't this capture `FileNotFoundError` only?

Reply:
Ah, damn you're right, thanks for catching that. Fixed